torch/csrc/byte_order.h

```diff
@@ -1,17 +1,18 @@
 #ifndef THP_BYTE_ORDER_H
 #define THP_BYTE_ORDER_H
 
-#include <cstdint>
-#include <cstddef>
 #include <THHalf.h>
 #include <c10/util/BFloat16.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <cstddef>
+#include <cstdint>
 
 enum THPByteOrder {
   THP_LITTLE_ENDIAN = 0,
   THP_BIG_ENDIAN = 1
 };
 
-THPByteOrder THP_nativeByteOrder();
+TORCH_API THPByteOrder THP_nativeByteOrder();
```

Comment:
It's not a function I'd expect to be part of the public API, but exposing it's symbol is probably harmless. Why do you need to `TORCH_API` this?

Reply:
Actually, I don't think this needs to be TORCH_API. This was probably left over from some build issues I was debugging.

Reply:
@ezyang How do you expose the symbol without using `TORCH_API`?

Reply:
TORCH_API is the correct way to expose the symbol.

While reviewing the rest of the patch, I realized that you have moved this file from the PyTorch Python library to libtorch. That's fine, but then we need to rename these symbols and put them in a namespace. Otherwise, they will pollute the global C++ namespace for our users.